### PR TITLE
correct the fallback order for runtime.json

### LINKED ISSUE #2891: Incorrect runtime.json processing
When restoring a package that has a runtime.json, we end up picking the least specific runtime ID, not the most specific runtime ID. As far as we can tell, this hasn’t produced any known critical issues, but it means we’re picking the wrong dependencies. For example, System.Net.Primitives has the following runtime.json:

```json
{
  "runtimes": {
    "win7": {
      "System.Net.Primitives": {
        "runtime.win7.System.Net.Primitives": "4.0.11-beta-23401"
      }
    },
    "osx.10.10": {
      "System.Net.Primitives": {
        "runtime.osx.10.10.System.Net.Primitives": "4.0.11-beta-23401"
      }
    },
    "unix": {
      "System.Net.Primitives": {
        "runtime.unix.System.Net.Primitives": "4.0.11-beta-23401"
      }
    }
  }
}
```

On `osx.10.10-x64`, we end up picking `runtime.unix.System.Net.Primitives`.

/cc @Eilon @muratg @davidfowl @DamianEdwards @danroth27 @BrennanConroy - May be a candidate for beta8. I'll send an email shortly ;)

### PR DESCRIPTION
**Pull request by anurse**
fixes #2891 

/cc @Eilon @muratg @davidfowl @DamianEdwards @danroth27 @BrennanConroy

### CODE REVIEW

File: src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs [hunk 1]
```diff
@@ -229,8 +229,8 @@ public RestoreCommand(FrameworkName fallbackFramework)
                     .Where(x => x.Severity == DiagnosticMessageSeverity.Error)
                     .Select(x => x.Message);
 
-                throw new InvalidOperationException(errorMessages.Any() ? 
-                    $"Errors occured when while parsing project.json:{Environment.NewLine}{string.Join(Environment.NewLine, errorMessages)}" : 
+                throw new InvalidOperationException(errorMessages.Any() ?
```
Comment (anurse) [inline src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs]: Just fallout from running Ctrl-K Ctrl-D :)

**Comment by anurse** [comment]
Travis only failed on OSX and it was because of a fix I made in dev that needs to be ported here.

**Comment by davidfowl** [comment]
Add a test. The fact that we have no coverage doesnt inspire confidence :smile:

**Comment by anurse** [comment]
Adding much more than a mocked-up unit test is going to take a significant effort, so it depends on when we need to get this in.

**Comment by anurse** [comment]
I'll build a unit test to start with. We can see what that looks like

**Comment by anurse** [comment]
Added a unit test.

**Comment by anurse** [comment]
ping @davidfowl

**Comment by davidfowl** [comment]
:shipit:

**Comment by anurse** [comment]
Merged

**Pull request closed without merging**